plugins/coreboot/fu-plugin-coreboot.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-device-metadata.h"
+#include "fu-device-private.h"
+#include "fu-plugin-coreboot.h"
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	/* make sure that flashrom plugin is ready to receive devices */
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+}
+
+static gboolean
+fu_plugin_get_guids (FuDevice *dev, FuPlugin *plugin, GError **error)
+{
+	const gchar *vendordefined[] = {
+		FU_HWIDS_KEY_MANUFACTURER "&"
+		FU_HWIDS_KEY_FAMILY "&"
+		FU_HWIDS_KEY_PRODUCT_NAME "&"
+		FU_HWIDS_KEY_PRODUCT_SKU "&"
+		FU_HWIDS_KEY_BIOS_VENDOR "&"
+		FU_HWIDS_KEY_BIOS_VERSION,
+
+		FU_HWIDS_KEY_MANUFACTURER "&"
+		FU_HWIDS_KEY_FAMILY "&"
+		FU_HWIDS_KEY_PRODUCT_NAME "&"
+		FU_HWIDS_KEY_BIOS_VENDOR "&"
+		FU_HWIDS_KEY_BIOS_VERSION,
+
+		FU_HWIDS_KEY_MANUFACTURER "&"
+		FU_HWIDS_KEY_FAMILY "&"
+		FU_HWIDS_KEY_PRODUCT_NAME "&"
+		FU_HWIDS_KEY_PRODUCT_SKU "&"
+		FU_HWIDS_KEY_BIOS_VENDOR,
+
+		FU_HWIDS_KEY_MANUFACTURER "&"
+		FU_HWIDS_KEY_FAMILY "&"
+		FU_HWIDS_KEY_PRODUCT_NAME "&"
+		FU_HWIDS_KEY_BIOS_VENDOR,
+	};
```

Comment:
Aren't some of these already specified HwID numbers? Can you just use those from fu-hwids?

Reply:
No they are not.